src/async_wrap.cc

```diff
@@ -189,6 +189,17 @@ void AsyncWrap::EmitAfter(Environment* env, double async_id) {
        env->async_hooks_after_function());
 }
 
+static double GetAssignedPromiseAsyncId(Environment* env,
+                                        Local<Promise> promise,
+                                        Local<Value> id_symbol) {
+  Local<Value> maybe_async_id;
+  if (!promise->Get(env->context(), id_symbol).ToLocal(&maybe_async_id)) {
+    return AsyncWrap::kInvalidAsyncId;
+  }
```

Comment:
Can we propagate the failure instead, i.e. change the return type to `v8::Maybe<double>` and return `Nothing<double>()` instead?

Reply:
Sure thing. I've changed the function, so it now returns `v8::Maybe<double>`.